### ISSUE TITLE
Likely bug in handling of header in h2o.conf

### ISSUE THREAD

**Issue by utrenkner** [h2o/h2o#3332]
I believe I have found a bug in how H2O handles headers, which are configured in h2o.conf: Adding or setting a header at host or path level under one host, removes all headers set at global level for other hosts.

I stumbled on this problem on one of our production servers. This is the simplest h2o.conf to explain the problem:
```
header.set: "x-frame-options: deny"

hosts:
  www.example1.test:
    listen:
      host: 127.0.0.1
      port: 80
    paths:
      "/":
        file.dir: "/usr/local/www/data/www.example1.test"
  www.example2.test:
    listen:
      host: 127.0.0.1
      port: 80
    paths:
      "/":
        file.dir: "/usr/local/www/data/www.example2.test"
```
Using this h2o.conf, both hosts deliver files with the header ``x-frame-options: deny``. Now, if I add or set any other header to one of the hosts, e.g. ``header.set: "Content-Type: text/html"``, then the the other host loses the ``x-frame-options: deny`` (and any other) header set at global level. 

If I use this h2o.conf

```
header.set: "x-frame-options: deny"

hosts:
  www.example1.test:
    listen:
      host: 127.0.0.1
      port: 80
    paths:
      "/":
        file.dir: "/usr/local/www/data/www.example1.test"
  www.example2.test:
    listen:
      host: 127.0.0.1
      port: 80
    header.set: "Content-Type: text/html"
    paths:
      "/":
        file.dir: "/usr/local/www/data/www.example2.test"
```
files at www.example2.test are delivered with ``x-frame-options: deny`` and ``content-type: text/html``. But files at www.example1.test are delivered without ``x-frame-options: deny``.

I tried various headers at global, host and path level. I tried adding or setting additional headers at host www.example1.test and www.example2.test: I always encountered this problem.

Note, I used the official FreeBSD port by @dch which uses a snapshot of 7 November 2023 (at b15937e).

**Comment by utrenkner** [comment]
Just noticed: Any of the header-configurations at path level also eliminates headers set at host level. E.g. I set a certain header at host level, but for one specific path, I need to unset it. However, this leads all headers set at host level to disappear in all other paths of that host.

**Comment by kazuho** [comment]
Closed by #3353.

Also, the issue has been disclosed as a vulnerability; see https://github.com/h2o/h2o/security/advisories/GHSA-5m7v-cj65-h6pj. We apologize for the delay of the disclosure.

@utrenkner Thank you for contacting us.
